File: src/msa/web-service/portalapi/src/main/java/vn/com/kns/portalapi/application/service/shop/shop/dto/ShopDto.java
package vn.com.kns.portalapi.application.service.shop.shop.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import vn.com.kns.portalapi.application.service.category.contact.dto.ContactDto;
import vn.com.kns.portalapi.application.service.category.country.dto.CountryDto;
import vn.com.kns.portalapi.application.service.category.district.dto.DistrictDto;
import vn.com.kns.portalapi.application.service.category.province.dto.ProvinceDto;
import vn.com.kns.portalapi.core.model.BaseEntityDto;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ShopDto extends BaseEntityDto {
    private String code;
    private String name;
    private String phone;
    private String address;
    private Boolean isActive;

    private CountryDto country;
    private Long countryId;
    private ProvinceDto province;
    private Long provinceId;
    private DistrictDto district;
    private Long districtId;
//    private CountryDto country;
    private Long wardId;
//    private CountryDto country;
    private Long streetId;

    private List<ContactDto> contacts;
}

File: src/msa/main-service/authportal-service/src/app/shared/services/shop/shop/dto/shop-dto.ts
import {CountryDto} from '../../../category/country/dto/country-dto';
import {ProvinceDto} from '../../../category/province/dto/province-dto';
import {DistrictDto} from '../../../category/district/dto/district-dto';
import {ContactDto} from '../../../contact/dto/contact-dto';

export class ShopDto {
  id: number | null;
  name: string;
  code: string;
  phone: string;
  address: string;
  isActive: boolean;

  country: CountryDto = new CountryDto();
  countryId: number;
  province: ProvinceDto = new ProvinceDto();
  provinceId: number;
  district: DistrictDto = new DistrictDto();
  districtId: number;

  contacts: ContactDto[] = [];

  constructor(id?: number | null, name?: string, code?: string) {
    this.id = id;
    this.name = name;
    this.code = code;
  }
}
